lib/zypper.rb

```diff
@@ -86,10 +91,12 @@ def version
     [found[1].to_i, found[2].to_i, found[3].to_i] if found
   end
 
-  private
-
   def call_zypper(*args)
-    cmd = ["zypper"]
+    cmd = if args.last.is_a?(Hash) && args.last[:sudo]
```

Comment:
From reading the code I'd think that you have to pop the last elements in the positive case because it would otherwise end up in the command. But why doesn't the test fail then? ...